src/main/java/org/mifos/connector/gsma/transfer/CorrelationIDStore.java

```diff
@@ -23,4 +25,16 @@ public void addMapping(String serverCorrelation, String clientCorrelation) {
     public String getClientCorrelation (String serverCorrelation) {
         return correlation.get(serverCorrelation);
     }
+
+    public Stream<String> getServerCorrelations(String clientCorrelation) {
+        return correlation
+                .entrySet()
+                .stream()
+                .filter(entry -> clientCorrelation.equals(entry.getValue()))
+                .map(Map.Entry::getKey);
+    }
```

Comment:
@SubhamPramanik Are these correlationIds fetched from Zeebe variables or from in memory? Just want to make sure that these IDs are not lost in case of downtime of GSMA connector.

Reply:
Currently, this is in memory. We need to discuss this.